src/Azure.Functions.Cli/Actions/LocalActions/InitAction.cs

```diff
@@ -157,8 +162,10 @@ private async Task InitFunctionAppProject()
             }
             else
             {
-                await InitLanguageSpecificArtifacts(workerRuntime, language);
+                bool managedDependenciesOption = ResolveManagedDependencies(workerRuntime, ManagedDependencies);
+                await InitLanguageSpecificArtifacts(workerRuntime, language, managedDependenciesOption);
```

Comment:
Should we move the call to `ResolveManagedDependencies` in  `InitLanguageSpecificArtifacts` because it should be only done and used for `powershell` artifacts? `ResolveManagedDependencies` can update `ManagedDependencies` flag if needed.

This way `WriteHostJson` can also use `ManagedDependencies` directly and won't need to expect a flag.

Not a big deal. If you prefer it this way, that's fine with me too.

Reply:
If managed dependencies is enable, we need to do two things: create a requirements.psd1 file and add an entry to host.json. This the reason why we resolved it before calling into InitLanguageSpecificArtifacts and WriteHostJson. Is there a better way to do this?